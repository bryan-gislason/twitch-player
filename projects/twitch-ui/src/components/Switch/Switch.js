import { useTheme } from "@twitch-player/themes/dist/hooks";
import React, { memo } from "react";
import { Switch } from "react-native";

const CustomSwitch = ({ ...props }) => {
  const [{ colors }] = useTheme();

  return (
    <Switch
      {...props}
      thumbColor={colors.switchThumb}
      trackColor={colors.switchTrack}
      activeThumbColor={colors.switchActiveThumb}
      activeTrackColor={colors.switchActiveTrack}
    />
  );
};

export default memo(CustomSwitch);
