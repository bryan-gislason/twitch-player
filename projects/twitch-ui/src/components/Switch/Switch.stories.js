import { action } from "@storybook/addon-actions";
import { storiesOf } from "@storybook/react";
import React from "react";
import Switch from ".";

storiesOf("Static Components|Switch", module)
  .add("off", () => <Switch value={false} onValueChange={action("changed")} />)
  .add("on", () => <Switch value={true} onValueChange={action("changed")} />)
  .add("disabled", () => <Switch value={false} disabled={true} />);
