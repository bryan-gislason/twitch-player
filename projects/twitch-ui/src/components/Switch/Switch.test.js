import React from "react";
import { render } from "../../testing";
import Switch from "./Switch";

describe("Switch", () => {
  let instance;

  beforeEach(() => {
    instance = render(
      <Switch values={{ foo: "Foo", bar: "Bar" }} testID="picker" />
    );
  });

  it("renders a switch", () => {
    expect(instance.getByTestId("picker")).toMatchSnapshot();
  });
});
