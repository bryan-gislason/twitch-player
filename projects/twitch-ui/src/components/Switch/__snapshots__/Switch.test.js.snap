// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders a switch 1`] = `
<div
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-cursor-1loqt21 rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-userSelect-lrvibr rn-zIndex-1lgpqti"
  data-testid="picker"
  style="height: 20px; width: 40px;"
>
  <div
    class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-bottom-1p0dtai rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-1dernwh rn-left-1d2f490 rn-marginTop-1t01tom rn-marginRight-lchren rn-marginBottom-1qahzrx rn-marginLeft-1jj8364 rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-u8s1d rn-right-zchlnj rn-top-ipm5af rn-transitionDuration-13tjlyg rn-width-13qz1uu rn-zIndex-1lgpqti"
    style="background-color: rgb(218, 216, 222); border-top-left-radius: 10px; border-top-right-radius: 10px; border-bottom-right-radius: 10px; border-bottom-left-radius: 10px;"
  />
  <div
    class="rn-alignItems-1oszu61 rn-alignSelf-k200y rn-borderTopLeftRadius-jt3ufn rn-borderTopRightRadius-1e868j9 rn-borderBottomRightRadius-ujv9e3 rn-borderBottomLeftRadius-1hakmuk rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxShadow-1ewcgjf rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-left-1fe0xdi rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-transform-emqnss rn-transitionDuration-13tjlyg rn-zIndex-1lgpqti"
    style="background-color: rgb(250, 249, 250); height: 20px; width: 20px;"
  />
  <input
    class="rn-bottom-1p0dtai rn-cursor-1ei5mc7 rn-height-1pi2tsx rn-left-1d2f490 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-opacity-1272l3b rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-u8s1d rn-right-zchlnj rn-top-ipm5af rn-width-13qz1uu"
    data-focusable="true"
    type="checkbox"
  />
</div>
`;
